### PR TITLE
[DEVHAS-492] Test for Component Devfile with Private Parent references

### PR DESCRIPTION
**Pull request by maysunfaisal**
### What does this PR do?:

- Adds in tests for Component Devfile with Private Parent references

### Which issue(s)/story(ies) does this PR fixes:

https://issues.redhat.com/browse/DEVHAS-492

### PR acceptance criteria:

- [x] Unit/Functional tests

  

- [ ] Documentation 

   

- [ ] Client Impact

  

### How to test changes / Special notes to the reviewer:
Tests should pass

### CODE REVIEW

**Comment by maysunfaisal** [comment]
Once https://github.com/devfile/library/pull/192 is merged, this PR go module should be updated to pull in the latest devfile/library

File: controllers/component_controller.go [hunk 1]
```diff
@@ -485,7 +498,10 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		// If the model already exists, see if fields have been updated
 		log.Info(fmt.Sprintf("Checking if the Component has been updated %v", req.NamespacedName))
 
-		// Parse the Component Devfile
+		// Parse the Component CR Devfile
+		// Not necessary to pass in a Token or DevfileUtils client to the parser here since the devfileBytes has:
```
Comment (kim-tsao) [inline controllers/component_controller.go:502]: same comment as above

File: controllers/component_controller.go [hunk 2]
```diff
@@ -224,7 +224,10 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	for _, condition := range component.Status.Conditions {
 		if forceGenerateGitopsResource || (condition.Type == "GitOpsResourcesGenerated" && condition.Reason == "GenerateError" && condition.Status == metav1.ConditionFalse) {
 			log.Info(fmt.Sprintf("Re-attempting GitOps generation for %s", component.Name))
-			// Parse the Component Devfile
+			// Parse the Component CR Devfile
+			// Not necessary to pass in a Token or DevfileUtils client to the parser here since the devfileBytes has:
```
[the remark this answers is not in the document]
Comment (maysunfaisal) [reply controllers/component_controller.go:228]: May do that, I need to test the change with a live cluster just for my sake.

File: controllers/component_controller.go (hunk 1 above)
Comment (maysunfaisal) [reply controllers/component_controller.go:502]: sniff tested on a live cluster after removing these tokens from the invocation and it worked well..

File: controllers/component_controller.go [hunk 3]
```diff
@@ -411,8 +418,11 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 				return ctrl.Result{}, err
 			}
 		} else {
-			// Parse the Component Devfile
 			log.Info(fmt.Sprintf("Parsing Devfile from the Devfile bytes %v... %v", len(devfileBytes), req.NamespacedName))
+			// Parse the Component CR Devfile
+			// Not necessary to pass in a Token or a DevfileUtils client to the parser here on devfileBytes, since:
```
[the remark this answers is not in the document]
Comment (maysunfaisal) [reply controllers/component_controller.go:423]: sniff tested on a live cluster after removing these tokens from the invocation and it worked well..

File: controllers/component_controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (maysunfaisal) [reply controllers/component_controller.go:228]: sniff tested on a live cluster after removing these tokens from the invocation and it worked well..

**Pull request merged**